Modules/Core/Common/include/itkThreadSupport.h

```diff
@@ -49,12 +49,12 @@ using MutexType = pthread_mutex_t;
 using FastMutexType = pthread_mutex_t;
 using ThreadFunctionType = void * (*)(void *);
 using ThreadProcessIdType = pthread_t;
+#if !defined(ITK_LEGACY_REMOVE)
 constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID = 0;
+#endif
 using ITK_THREAD_RETURN_TYPE = void *;
 
-/* ITK_THREAD_RETURN_DEFAULT_VALUE is from a c library, and always needs to be NULL, not nullptr */
```

Comment:
I sure did. :) Hence this being WIP. It compiles on my computer as nullptr, I don't understand why it wouldn't elsewhere... this is a C++ file, not C....

Reply:
If I remember correctly, there was an incantation of compiler/build settings that issued a warning about mismatched types.    I think it was a Windows compiler (because I don't usually have easy access to test there).

This comment and the NULL have bothered me for years.  I wish I had written a more comprehensive commit message that documented the exact variant that prompted this comment.

Reply:
Ah, well if it was just a warning, and years ago... we may be better off with nullptr at this point, since newer compilers warn about *not* using nullptr :)  And, again, it's a C++ file, so I just don't see how nullptr can be a problem...

Reply:
So CI is green with the nullptr change...